java/ql/test/library-tests/frameworks/android/widget/TestWidgetKt.kt

```diff
@@ -0,0 +1,16 @@
+ import android.text.Editable
+
+ class TestWidget {
+
+    fun source() : Editable? { return null }
+    fun sink(sink : String) {}
+
+    fun test() {
+        val t = source()
+        sink(t.toString()); // $ hasTaintFlow
+
+        val t2 : Any? = source()
+        sink(t2.toString()); // $ MISSING: hasTaintFlow
```

Comment:
Why does it still fail?

Reply:
`t2`'s type is `Object` because of the `Any?` declaration. So the taint step works for `Editable?` but not for `Any?`.

This is also causing other issues, e.g. `toString`'s declaring type is `Any` in both cases, which interferes with our `Method::override` logic.

Reply:
I believe that's consistent with the Java extractor -- if the receiver type doesn't supply an override then MethodAccess.getMethod() points to the inherited method.